### PR TITLE
feat: E2EE warnings on search and audit panel

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes (including videos or screenshots)

While end-to-end encryption in Rocket.Chat provides enhanced privacy and security for communications, it inherently restricts certain functionalities, such as auditing messages, searching message content, and interacting with bots within encrypted rooms. Users may not be aware of these limitations, which can lead to confusion and a mismatch between user expectations and the platform's capabilities. Incorporating clear disclaimers into the user interface will help set accurate expectations and improve user satisfaction by transparently communicating the trade-offs involved with encryption.

image

image

## Further comments

[E2EE2-10](https://rocketchat.atlassian.net/browse/E2EE2-10)

[E2EE2-10]: https://rocketchat.atlassian.net/browse/E2EE2-10?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dougfabris — CHANGES_REQUESTED** [review]
I'm thinking we're adding unnecessary complexity in the user experience. Basically we're adding the support for finding the encrypted rooms and when they select, we warn them they can't proceed with what they want (audit feature). If the issue is that users are being frustrated because they don't know they can't use these features for e2e encryption we're failing on the explanation of our features. We should look for an alternative to warn them such as displaying these callouts based on e2e being enabled on the workspace, saying they won't be able to find the encrypted channels for example. Kinda similar what you're doing on search messages.

And this change definitely isn't a `chore`

**Comment by yash-rajpal** [comment]
Thanks for the review doug, and I'm always having hard time figuring out if something is a `chore` or not 🤔 

But the idea is, encrypted rooms are still auditable, and encrypted rooms might contain some un-encrypted messages which can be audited. The warning is just to let the users know about the limitations. 

If we stop showing encrypted rooms on audit panel then the admin won't be able to see the un-encrypted messages which could still be audited.

File: packages/i18n/src/locales/en.i18n.json [hunk 1]
```diff
@@ -1938,8 +1938,14 @@
...
   "Encrypted": "Encrypted",
   "Encrypted_channel_Description": "Messages are end-to-end encrypted, search will not work and notifications may not show message content",
+  "Encrypted_content_cannot_be_searched": "Encrypted content cannot be searched.",
   "Encrypted_key_title": "Click here to disable end-to-end encryption for this channel (requires e2ee-permission)",
   "Encrypted_message": "Encrypted message",
+<<<<<<< Updated upstream
+  "Encrypted_RoomType": "Encrypted {{roomType}}",
+=======
+  "Encrypted_message_preview_unavailable": "Encrypted message, preview unavailable",
+>>>>>>> Stashed changes
```
Comment (hugocostadev) [inline packages/i18n/src/locales/en.i18n.json:1948]: I guess it has a error here

**Review by hugocostadev — CHANGES_REQUESTED** [review]
fix errors and implement tests

File: packages/i18n/src/locales/en.i18n.json (hunk 1 above)
Comment (yash-rajpal) [reply packages/i18n/src/locales/en.i18n.json:1948]: How'd that get here 🤔 

Will remove

**Comment by hugocostadev** [comment]
Tests requirement dismissed due to the nature of this feature, does not worth to do e2e tests to check warning

**Review by hugocostadev — DISMISSED** [review]

File: .changeset/friendly-months-attack.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@rocket.chat/i18n': patch
+'@rocket.chat/meteor': patch
```
Comment (MarcosSpessatto) [inline .changeset/friendly-months-attack.md:3]:
```suggestion
'@rocket.chat/i18n': minor
'@rocket.chat/meteor': minor
```

File: apps/meteor/client/lib/getRoomTypeTranslation.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import type { IRoom } from '@rocket.chat/core-typings';
+
+import { t } from '../../app/utils/lib/i18n';
+
+export const getRoomTypeTranslation = (type: IRoom['t']) => {
```
Comment (MarcosSpessatto) [inline apps/meteor/client/lib/getRoomTypeTranslation.ts:5]: What about adding a unit test to cover this function?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/client/lib/getRoomTypeTranslation.ts (hunk 3 above)
Comment (yash-rajpal) [reply apps/meteor/client/lib/getRoomTypeTranslation.ts]: Aligned this with Marcos, the function is pretty basic and simple, doesn't need a test for this function.

**Review by hugocostadev — APPROVED** [review]

File: apps/meteor/client/components/RoomAutoComplete/RoomAutoComplete.tsx [hunk 4]
```diff
@@ -12,7 +13,11 @@ const generateQuery = (
 	selector: string;
 } => ({ selector: JSON.stringify({ name: term }) });
 
-type RoomAutoCompleteProps = Omit<ComponentProps<typeof AutoComplete>, 'filter'> & { scope?: 'admin' | 'regular' };
+type RoomAutoCompleteProps = Omit<ComponentProps<typeof AutoComplete>, 'filter'> & {
+	scope?: 'admin' | 'regular';
+	renderRoomIcon?: (props: { encrypted: IRoom['encrypted']; type: IRoom['t'] }) => ReactElement | null;
```
Comment (tassoevan) [inline apps/meteor/client/components/RoomAutoComplete/RoomAutoComplete.tsx:18]: Assume this behavior will be absorbed by the component if the design team agrees, making the property pointless.

**Review by tassoevan — APPROVED** [review]

File: apps/meteor/client/components/RoomAutoComplete/RoomAutoComplete.tsx (hunk 4 above)
Comment (yash-rajpal) [reply apps/meteor/client/components/RoomAutoComplete/RoomAutoComplete.tsx:18]: Yes, incase we plan to show this in other places as well, this behavior will be implemented by the component itself. Thanks Tasso!

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**